core/src/main/java/org/elasticsearch/index/search/QueryStringQueryParser.java

```diff
@@ -380,28 +472,28 @@ protected Query getRangeQuery(String field, String part1, String part2,
     private Query getRangeQuerySingle(String field, String part1, String part2,
                                       boolean startInclusive, boolean endInclusive, QueryShardContext context) {
         currentFieldType = context.fieldMapper(field);
-        if (currentFieldType != null) {
-            try {
-                Analyzer normalizer = forceAnalyzer == null ? queryBuilder.context.getSearchAnalyzer(currentFieldType) : forceAnalyzer;
-                BytesRef part1Binary = part1 == null ? null : normalizer.normalize(field, part1);
-                BytesRef part2Binary = part2 == null ? null : normalizer.normalize(field, part2);
-                Query rangeQuery;
-                if (currentFieldType instanceof DateFieldMapper.DateFieldType && timeZone != null) {
-                    DateFieldMapper.DateFieldType dateFieldType = (DateFieldMapper.DateFieldType) this.currentFieldType;
-                    rangeQuery = dateFieldType.rangeQuery(part1Binary, part2Binary,
-                        startInclusive, endInclusive, timeZone, null, context);
-                } else {
-                    rangeQuery = currentFieldType.rangeQuery(part1Binary, part2Binary, startInclusive, endInclusive, context);
-                }
-                return rangeQuery;
-            } catch (RuntimeException e) {
-                if (lenient) {
-                    return null;
-                }
-                throw e;
+        if (currentFieldType == null) {
+            return new MatchNoDocsQuery("unknown field " + field);
+        }
+        try {
+            Analyzer normalizer = forceAnalyzer == null ? queryBuilder.context.getSearchAnalyzer(currentFieldType) : forceAnalyzer;
+            BytesRef part1Binary = part1 == null ? null : normalizer.normalize(field, part1);
+            BytesRef part2Binary = part2 == null ? null : normalizer.normalize(field, part2);
+            Query rangeQuery;
+            if (currentFieldType instanceof DateFieldMapper.DateFieldType && timeZone != null) {
+                DateFieldMapper.DateFieldType dateFieldType = (DateFieldMapper.DateFieldType) this.currentFieldType;
+                rangeQuery = dateFieldType.rangeQuery(part1Binary, part2Binary,
+                    startInclusive, endInclusive, timeZone, null, context);
+            } else {
+                rangeQuery = currentFieldType.rangeQuery(part1Binary, part2Binary, startInclusive, endInclusive, context);
+            }
+            return rangeQuery;
+        } catch (RuntimeException e) {
+            if (lenient) {
+                return null;
```

Comment:
should this return a `MatchNoDocsQuery` like the other ones where an exception is thrown?